dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+        for k, v in metadata['dlhub']['dependencies']['python'].items():
+            dependencies.append("{0}=={1}".format(k, v))
+    except:
+        # There are no python dependencies
+        pass
+
+    model_location = None
+    if 'transfer_method' in metadata['dlhub']:
+        if 'S3' in metadata['dlhub']['transfer_method']:
+            model_location = metadata['dlhub']['transfer_method']['S3']
+        elif 'POST' in metadata['dlhub']['transfer_method']:
+            model_location = metadata['dlhub']['transfer_method']['path']
+    if 'repository' in metadata:
+        model_location = metadata['repository']
+    cs = ContainerSpec(
+        name=metadata['dlhub']['shorthand_name'],
+        pip=dependencies,
+        python_version="3.7",
+        payload_url=model_location,
+        conda=[],
+    )
+
+    return cs
+
+
+def search_ingest(task, header):
+    """
+    Ingest the servable data into a Globus Search index.
+
+    Args:
+        task (dict): the task description.
+    """
+    logger.debug("Ingesting servable into Search.")
+
+#    idx = "dlhub"
+    # idx = '847c9105-18a0-4ffb-8a71-03dd76dfcc9d'
+    iden = "https://dlhub.org/servables/{}".format(task['dlhub']['id'])
+    # index = mdf_toolbox.translate_index(idx)
+
+    ingestable = task
+    d = [convert_dict(ingestable, str)]
+
+    glist = []
+    visible_to = task['dlhub'].get('visible_to', ['public'])
+
+    # Add public so it isn't an empty list
+    if len(visible_to) == 0:
+        visible_to = ['public']
+
+    for document in d:
+        gmeta_entry = mdf_toolbox.format_gmeta(document, visible_to, iden)
+        glist.append(gmeta_entry)
+    gingest = mdf_toolbox.format_gmeta(glist)
+
+    logger.info("ingesting to search")
+    logger.info(gingest)
+
+    GLOBUS_SEARCH_WRITER_LAMBDA = "https://7v5g6s33utz4l7jx6dkxuh77mu0cqdhb.lambda-url.us-east-2.on.aws/"
+
+    # Get the authorization header token (string for the headers dict)
+    #header = self.authorizer.get_authorization_header()
+
+    http_response = requests.post(
+        GLOBUS_SEARCH_WRITER_LAMBDA,
+        headers={"Authorization": header},
+        json={'document_file': gingest})
+    if http_response.status_code != 200:
+        raise Exception(http_response.text)
+    logger.info("Ingestion of {} to DLHub servables complete".format(iden))
+
+
+def mint_identifier(task):
+    """Mint a new identifier and return it."""
+    identifiers_namespace = '1EGOGHSs9RAtq'
+    identifier = None
+    logger.debug("Creating identifier")
+    try:
+        # client = identifiers_client(config)
+        serv_location = "https://%s%s" % ("dlhub.org/servables/", task['dlhub']['id'])
+        # safe encode the location
+        serv_location = urllib.parse.quote_plus(serv_location, safe='')
+        # can't be too safe!
+        serv_location = urllib.parse.quote_plus(serv_location, safe='')
+        # Create the petrel link
+        serv_location = f'https://petreldata.net/dlhub/detail/{serv_location}'
```

Comment:
@blaiszik I assume this should be eagle, not petrel, right? or does the domain resolve to the same location?

Reply:
per discussion in Slack, this is moot since we're axing this function.